packages/trader/src/AppV2/Components/OnboardingGuide/onboarding-guide.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { Modal } from '@deriv-com/quill-ui';
+import { useLocalStorageData } from '@deriv/hooks';
+import { Localize } from '@deriv/translations';
+import GuideContainer from './guide-container';
+import OnboardingVideo from './onboarding-video';
+
+type TOnboardingGuideProps = {
+    type?: 'trade_page' | 'positions_page';
+};
+
+const OnboardingGuide = ({ type = 'trade_page' }: TOnboardingGuideProps) => {
+    const [is_modal_open, setIsModalOpen] = React.useState(false);
+    const [should_run_guide, setShouldRunGuide] = React.useState(false);
+    const guide_timeout_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const is_button_clicked_ref = React.useRef(false);
+
+    const [guide_dtrader_v2, setGuideDtraderV2] = useLocalStorageData<boolean>(`guide_dtrader_v2_${type}`, false);
+    const is_trade_page_guide = type === 'trade_page';
+
+    const onFinishGuide = React.useCallback(() => {
+        setShouldRunGuide(false);
+        setGuideDtraderV2(true);
+    }, [setGuideDtraderV2]);
+
+    const onGuideSkip = () => {
+        if (is_button_clicked_ref.current) return;
+        onFinishGuide();
+        setIsModalOpen(false);
+    };
+
+    const onGuideStart = () => {
+        is_button_clicked_ref.current = true;
+        setShouldRunGuide(true);
+        setIsModalOpen(false);
+    };
+
+    const modal_content = {
+        image: is_trade_page_guide ? <OnboardingVideo /> : <div className='video-placeholder' />,
```

Comment:
just a suggestion , rather then multiple ternary for same boolean can we do something like 
```

const modal_content = {
    image: <div className='video-placeholder' />,
    title: <Localize i18n_default_text='View your positions' />,
    ...(is_trade_page_guide && {
        image: <OnboardingVideo />,
.....
    }),
};

```